pkg/secretless/entrypoint/entrypoint.go

```diff
@@ -50,11 +50,17 @@ func StartSecretless(params *SecretlessOptions) {
 
 	// Prepare Service
 	secretless := proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
-	signal.StopOnExitSignal(secretless)
 
 	handlePerformanceProfiling(params.ProfilingMode)
 
-	secretless.Start()
+	err = secretless.Start()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	return func() {
+		signal.StopOnExitSignal(secretless)
+	}
```

Comment:
This patterns seems overly complicated for what it's doing. Why couldn't we just wait forever here rather than in the caller? also, secretless should implement `Stop` already iirc.